### PR TITLE
lib/config: Change folder type attribute to a FolderType type

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

Followup to #3029

### CODE REVIEW

File: lib/config/foldertype.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func (t FolderType) String() string {
+	switch t {
+	case FolderTypeReadWrite:
+		return "readwrite"
+	case FolderTypeReadOnly:
+		return "readonly"
+	default:
+		return "unknown"
+	}
```
Comment (AudriusButkevicius) [inline lib/config/foldertype.go]: Suggest we panic

File: lib/config/foldertype.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func (t *FolderType) UnmarshalText(bs []byte) error {
+	switch string(bs) {
+	case "readwrite":
+		*t = FolderTypeReadWrite
+	case "readonly":
+		*t = FolderTypeReadOnly
+	default:
+		*t = FolderTypeReadWrite
```
Comment (AudriusButkevicius) [inline lib/config/foldertype.go]: Here too

File: lib/config/folderconfiguration.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ type FolderDeviceConfiguration struct {
 	DeviceID protocol.DeviceID `xml:"id,attr" json:"deviceID"`
 }
 
-func NewFolderConfiguration(id, path, foldertype string) FolderConfiguration {
+func NewFolderConfiguration(id, path string, foldertype FolderType) FolderConfiguration {
 	f := FolderConfiguration{
```
Comment (AudriusButkevicius) [inline lib/config/folderconfiguration.go]: I think this is used only in one place, so defaulting to rw is fine, and we couldget rid of the arg.

**Comment by calmh** [comment]
I see no reason to panic in the cases above; there is a possibility of returning an error (not sure what happens then, the config load fails somehow I guess) though but we haven't historically done that for these things, i.e. compare PullOrder.

**Comment by AudriusButkevicius** [comment]
@st-review merge

lib/config: Change folder type attribute to a FolderType type

**Comment by st-review** [comment]
:clock1: Build status is `pending`. I'll wait to see if this becomes successful and then merge!

**Comment by st-review** [comment]
:ok_hand: Merged as 09832abe50ba08664b900a865cd6e8fa23ff636e. Thanks, @calmh!

**Pull request closed without merging**